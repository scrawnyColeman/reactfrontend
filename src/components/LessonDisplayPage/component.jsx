import React from 'react';
import styled from 'styled-components';
import { colours } from '../../constants/styles';

const Container = styled.div`
    display: grid;
    grid-template-rows: 1fr 1fr;
    grid-gap: 8px;
    padding 16px;
    border: 2px solid ${colours.primary}
`;

const StyledHead = styled.div``;
const StyledTitle = styled.div`
    font-size: 2rem;
    display: flex;
    justify-content: space-between;
`;
const StyledDesc = styled.div`
    font-size: 1.5rem;
    color: ${colours.secondary}
    margin-top: 5vh;
    text-align: left;
`;
const StyledBody = styled.div`
    text-align: left;
`;

export const StyledListItem = styled.li`
    margin: 4px;
    color: ${colours.secondary};
`;

const LessonDisplayPage = ({ title, language, description, recommendedLessons }) => (
    <Container>
        <StyledHead>
            <StyledTitle>
                <span>{title}</span>
                <span>[{language.language}]</span>
            </StyledTitle>
            <StyledDesc>{description}</StyledDesc>
        </StyledHead>
        <StyledBody>
            {recommendedLessons.length > 0 ? (
                <div>Before beginning this course, it is adviced that you have knowledge of the following:</div>
            ) : (
                <div>
                    There are no prerequisities for this lesson! Good luck and don't be afraid to visit the forum for
                    help.
                </div>
            )}
            {recommendedLessons.length > 0 &&
                recommendedLessons.map(
                    (prereq) => prereq && <StyledListItem key={prereq.id}>{prereq.title}</StyledListItem>,
                )}
        </StyledBody>
    </Container>
);

export default LessonDisplayPage;
