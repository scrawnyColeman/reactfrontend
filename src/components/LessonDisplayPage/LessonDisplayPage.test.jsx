import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import LessonDisplayPage, { StyledListItem } from './component';

Enzyme.configure({ adapter: new Adapter() });

describe('LessonDisplayPage component', () => {
    const title = 'Lesson title';
    const language = { language: 'Javascript' };
    const description = 'JS lesson';
    const recommendedLessons = [
        { title: 'Java', id: '1' },
        { title: 'Recursion', id: '2' },
    ];

    test('component matches snapshot with recommended lessons', () => {
        const wrapper = shallow(
            <LessonDisplayPage
                title={title}
                language={language}
                description={description}
                recommendedLessons={recommendedLessons}
            />,
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(StyledListItem).length).toBe(2);
    });

    test('component matches snapshot with no recommended lessons', () => {
        const wrapper = shallow(
            <LessonDisplayPage title={title} language={language} description={description} recommendedLessons={[]} />,
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(StyledListItem).length).toBe(0);
    });
});
