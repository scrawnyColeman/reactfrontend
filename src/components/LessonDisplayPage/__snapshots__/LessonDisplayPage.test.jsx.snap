// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LessonDisplayPage component component matches snapshot with no recommended lessons 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <span>
        Lesson title
      </span>
      <span>
        [
        Javascript
        ]
      </span>
    </styled.div>
    <styled.div>
      JS lesson
    </styled.div>
  </styled.div>
  <styled.div>
    <div>
      There are no prerequisities for this lesson! Good luck and don't be afraid to visit the forum for help.
    </div>
  </styled.div>
</styled.div>
`;

exports[`LessonDisplayPage component component matches snapshot with recommended lessons 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <span>
        Lesson title
      </span>
      <span>
        [
        Javascript
        ]
      </span>
    </styled.div>
    <styled.div>
      JS lesson
    </styled.div>
  </styled.div>
  <styled.div>
    <div>
      Before beginning this course, it is adviced that you have knowledge of the following:
    </div>
    <styled.li
      key="1"
      onClick={[Function]}
    >
      Java
    </styled.li>
    <styled.li
      key="2"
      onClick={[Function]}
    >
      Recursion
    </styled.li>
  </styled.div>
</styled.div>
`;
